app/src/main/java/com/example/chrono/main/timer/CircuitCreate.kt

```diff
@@ -0,0 +1,187 @@
+            setResult(Activity.RESULT_CANCELED)
+            finish()
+        }
+
+        bind!!.saveButton.setOnClickListener {
+            if (validateInputs()) {
+                saveCircuit()
+            }
+        }
+
+        bind!!.addSet.setOnClickListener {
+            addSet()
+        }
+        bind!!.minusSet.setOnClickListener {
+            minusSet()
+        }
+        bind!!.addWork.setOnClickListener {
+            addWork()
+        }
+        bind!!.minusWork.setOnClickListener {
+            minusWork()
+        }
+        bind!!.addRest.setOnClickListener {
+            addRest()
+        }
+        bind!!.minusRest.setOnClickListener {
+            minusRest()
+        }
+    }
+
+    private fun saveCircuit() {
+        val circuit = CircuitObject()
+        circuit.name = bind!!.nameInput.editText!!.text.toString()
+        circuit.sets = bind!!.setNum.text.toString().toInt()
+        circuit.work = bind!!.setWorkTime.text.toString().toInt()
+        circuit.rest = bind!!.setRestTime.text.toString().toInt()
+
+        // Save circuit in Shared Preferences
+        val circuits: CircuitsObject? = PreferenceManager.get<CircuitsObject>("CIRCUITS")
+        circuits!!.circuits!!.add(circuit)
+        PreferenceManager.put(circuits, circuits.key)
+
+        Log.i("prefs", "" + PreferenceManager.get("CIRCUITS"))
+        setResult(Activity.RESULT_OK)
+        finish()
+    }
+
+    private fun validateInputs(): Boolean {
+        if (bind!!.nameInput.isEmpty() or (bind!!.nameInput.editText!!.text.toString() == "")) {
+            Toast.makeText(this, "Please enter a circuit name", Toast.LENGTH_SHORT).show()
+            return false
+        }
+        if (!(bind!!.setWorkTime.text.toString().matches(("^[1-9]\\d*\$").toRegex()))) {
+            Toast.makeText(this, "Number of sets must be greater than 0", Toast.LENGTH_SHORT).show()
+            return false
+        }
+        if (!(bind!!.setWorkTime.text.toString().matches(("^[1-9]\\d*\$").toRegex()))) {
+            Toast.makeText(this, "Invalid Rest Time", Toast.LENGTH_SHORT).show()
+            return false
+        }
+        if (!(bind!!.setRestTime.text.toString().matches(("^[1-9]\\d*\$").toRegex()))) {
+            Toast.makeText(this, "Invalid Rest Time", Toast.LENGTH_SHORT).show()
+            return false
+        }
+        return true
+    }
+
+    private fun addSet() {
+        if (bind!!.setNum.text.toString() == "") {
+            bind!!.setNum.setText("1")
+        } else {
+            bind!!.setNum.setText((bind!!.setNum.text.toString().toInt() + 1).toString())
+        }
+    }
+
+    private fun minusSet() {
+        if (bind!!.setNum.text.toString() == "") {
+            Toast.makeText(
+                this,
+                "Can't have a negative number of sets..? " + ("\uD83E\uDD14"),
+                Toast.LENGTH_SHORT
+            ).show()
+        } else {
+            if (bind!!.setNum.text.toString().toInt() != 0) {
```

Comment:
if its 0, I think it can still pass this check. Probably want to use >0

Reply:
It's impossible to get to negative numbers anyways, so simpler to check for the 0. Try it out you'll get what I mean.